haystack/components/evaluators/document_mrr.py

```diff
@@ -67,17 +67,23 @@ def run(
 
         for ground_truth, retrieved in zip(ground_truth_documents, retrieved_documents):
             score = 0.0
+            flag = False
+            ground_truth_content = []
+
             for ground_document in ground_truth:
                 if ground_document.content is None:
                     continue
+                ground_truth_content.append(ground_document.content)
 
-                for rank, retrieved_document in enumerate(retrieved):
-                    if retrieved_document.content is None:
-                        continue
-
-                    if ground_document.content in retrieved_document.content:
-                        score = 1 / (rank + 1)
-                        break
+            for rank, retrieved_document in enumerate(retrieved):
+                if flag:
+                    break
+                if retrieved_document.content is None:
+                    continue
+                if retrieved_document.content in ground_truth_content:
+                    score = 1 / (rank + 1)
+                    flag = True
+                    break
```

Comment:
I'm not sure I get the logic: this is the only place where `flag` is set to `True`, but we break out of the loop that's the only place where `flag` is evaluated...

Reply:
You are right. The flag was used to break out of nested loops, but they were replaced so its not needed anymore. 

> I'm not sure I get the logic: this is the only place where `flag` is set to `True`, but we break out of the loop that's the only place where `flag` is evaluated...